// paypal-rest-sdk/lib/paypal-rest-sdk.js
/* Copyright 2015-2016 PayPal, Inc. */
"use strict";

var configuration = require('paypal-rest-sdk/lib/configure');
var api = require('paypal-rest-sdk/lib/api');

module.exports = function () {

    function configure(options) {
        api.configure(options);
    }

    function generateToken(config, cb) {
        api.generateToken(config, cb);
    }

    return {
        version: configuration.sdkVersion,
        configure: configure,
        configuration: configuration.default_options,
        generateToken: generateToken,
        payment: require('paypal-rest-sdk/lib/resources/Payment')(),
        sale: require('paypal-rest-sdk/lib/resources/Sale')(),
        refund: require('paypal-rest-sdk/lib/resources/Refund')(),
        authorization: require('paypal-rest-sdk/lib/resources/Authorization')(),
        capture: require('paypal-rest-sdk/lib/resources/Capture')(),
        order: require('paypal-rest-sdk/lib/resources/Order')(),
        payout: require('paypal-rest-sdk/lib/resources/Payout')(),
        payoutItem: require('paypal-rest-sdk/lib/resources/PayoutItem')(),
        billingPlan: require('paypal-rest-sdk/lib/resources/BillingPlan')(),
        billingAgreement: require('paypal-rest-sdk/lib/resources/BillingAgreement')(),
        creditCard: require('paypal-rest-sdk/lib/resources/CreditCard')(),
        invoice: require('paypal-rest-sdk/lib/resources/Invoice')(),
        invoiceTemplate: require('paypal-rest-sdk/lib/resources/InvoiceTemplate')(),
        openIdConnect: require('paypal-rest-sdk/lib/resources/OpenIdConnect')(),
        webProfile: require('paypal-rest-sdk/lib/resources/WebProfile')(),
        notification: require('paypal-rest-sdk/lib/resources/Notification')(),
        //entries below are deprecated but provided for compatibility with 0.* versions
        generate_token: generateToken,
        billing_plan: require('paypal-rest-sdk/lib/resources/BillingPlan')(),
        billing_agreement: require('paypal-rest-sdk/lib/resources/BillingAgreement')(),
        credit_card: require('paypal-rest-sdk/lib/resources/CreditCard')(),
        openid_connect: require('paypal-rest-sdk/lib/resources/OpenIdConnect')()
    };
};
